radio/bluetooth/bluetooth.go

```diff
@@ -76,20 +79,68 @@ func Disconnect(client ble.Client) error {
 	return nil
 }
 
-func Scan(id string, timeout time.Duration) (map[string]bool, error) {
+func WriteCharacteristic(client ble.Client, characteristic *ble.Characteristic, value []byte, noRsp bool) error {
+	err := make(chan error)
+	go func() {
+		err <- client.WriteCharacteristic(characteristic, value, noRsp)
+	}()
+
+	select {
+	case done := <-err:
+		return done
+	case <-time.After(shortTimeout):
+		return fmt.Errorf("Write characteristic timed out")
+	}
+}
+
+func ReadCharacteristic(client ble.Client, characteristic *ble.Characteristic) ([]byte, error) {
+	type Result struct {
+		Val []byte
+		Err error
+	}
+
+	result := make(chan Result)
+	go func() {
+		result <- func() Result {
+			val, err := client.ReadCharacteristic(characteristic)
+			return Result{val, err}
+		}()
+	}()
+
+	select {
+	case done := <-result:
+		return done.Val, done.Err
+	case <-time.After(shortTimeout):
+		return nil, fmt.Errorf("Read characteristic timed out")
+	}
+}
+
+func WriteDescriptor(client ble.Client, descriptor *ble.Descriptor, value []byte) error {
+	err := make(chan error)
+	go func() {
+		err <- client.WriteDescriptor(descriptor, value)
+	}()
+
+	select {
+	case done := <-err:
+		return done
+	case <-time.After(shortTimeout):
+		return fmt.Errorf("Write descriptor timed out")
+	}
+}
+
+func Scan(id string) (map[string]bool, error) {
 	devices := make(map[string]bool)
 	advChannel := make(chan ble.Advertisement)
-	ctx := ble.WithSigHandler(context.WithTimeout(context.Background(), timeout*time.Second))
+	ctx := ble.WithSigHandler(context.WithTimeout(context.Background(), longTimeout))
 
 	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case adv := <-advChannel:
-				if strings.EqualFold(adv.LocalName(), id) {
-					devices[adv.Address().String()] = true
-				}
+		select {
+		case <-ctx.Done():
+			return
+		case adv := <-advChannel:
+			if strings.EqualFold(adv.LocalName(), id) {
+				devices[adv.Address().String()] = true
 			}
```

Comment:
And here. I'm not sure which way is the accepted go indentation of closing braces, but you've used both styles in this file so I'd just make them consistent.